### PR TITLE
replace opis/closure with laravel/serializable-closure

### PR DESCRIPTION
**Pull request by ctrl-f5**
| Q             | A
| ------------- | ---
| Branch        | master for features and deprecations
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Documented?   | no

replace opis/closure with laralvel/serialzable-closure to support php 8.1

partially solves https://github.com/phpro/grumphp/issues/957

# New Task Checklist:

- [x] Are the dependencies added to the composer.json suggestions?
- [x] Is the doc/tasks.md file updated?
- [x] Are the task parameters documented?
- [x] Is the task registered in the tasks.yml file?
- [x] Does the task contains phpunit tests?
- [x] Is the configuration having logical allowed types?
- [x] Does the task run in the correct context?
- [x] Is the `run()` method readable?
- [x] Is the `run()` method using the configuration correctly?
- [ ] Are all CI services returning green?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,9 +22,9 @@
         "amphp/parallel-functions": "1.0",
```
Comment (veewee) [inline composer.json:22]: Maybe best to upgrade amp/paralllel-functions to 1.1 here?
Otherwise opis will still be used ?

Comment (ctrl-f5) [reply composer.json]: I had some issues running the tests locally, had the same errors as in the pipelines on my local machine, even on master. So I assumed something was wrong with my machine.

I can try upping `amphp/parallel-functions` to 1.1, but then we're back on the `The given data could not be serialized` error :confused:

Comment (veewee) [reply composer.json]: Yes, that's an error we need to get resolved in  this PR. It should work for parallel functions ^1.1

Some of the pipelines were indeed failing earlier today as well.
but in order to get this released in a trustworthy way, all actions should be green again

**Comment by ctrl-f5** [comment]
This issue was preventing us from moving to php8.1, but now we've also started seeing issues with grumphp on 8.0. (the same issue as in the checks in this MR.)

We have decided to just stop using grumphp on all our projects and packages and move the functionality to our `make` workflow, which already handled all other automations.

As a result I will not be putting any more effort in this MR.

**Comment by veewee** [comment]
Sad to hear @ctrl-f5!
Especially since it is all caused by dependencies and is pretty much out of our control :)

Anyways, Good luck with your projects and packages!

**Pull request closed without merging**